pkg/apis/kops/cluster.go

```diff
@@ -349,6 +352,18 @@ type ClusterSubnetSpec struct {
 	Type SubnetType `json:"type,omitempty"`
 }
 
+type EgressProxySpec struct {
+	HTTPProxy     HTTPProxySpec `json:"httpProxy,omitempty"`
```

Comment:
@johnzeringue here is an example of `EgressProxySpec` that is set for HTTP proxy.  Both you and @DerekV were working on the same problem at the same time.  I think a combination of both your and his PR would be awesome. 

Guys let me know if I can assist.